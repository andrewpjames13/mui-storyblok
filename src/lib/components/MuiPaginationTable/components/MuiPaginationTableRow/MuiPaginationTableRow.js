import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import { TableRow } from '@material-ui/core';
import MuiTableCell from '../../../MuiTable/components/MuiTableRow/components/MuiTableCell/MuiTableCell';

import StoryBlok from '../../../../utils/Storyblok';

const MuiPaginationTableRow = ({
  rootClass,
  content,
  hover,
  selected,
}) => {
  const components = {
    MuiTableCell,
  };

  const styles = StoryBlok.arrayToMuiStyles(rootClass);

  return (
    <TableRow
      className={styles.root}
      hover={hover}
      selected={selected}
    >
      {content.map((item, index) => createElement(
        components[item.component],
        Object.assign(item, { key: index }),
      ))}
    </TableRow>
  );
};

export default MuiPaginationTableRow;

MuiPaginationTableRow.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui prop: true | false
   * If true, the table row will shade on hover.
   */
  hover: PropTypes.bool,
  /**
   * mui prop: true | false
   * If true, the table row will have the selected shading.
   */
  selected: PropTypes.bool,
  /**
   * Content passed to render
   * components:
   * MuiTableCell
   */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiPaginationTableRow.defaultProps = {
  hover: false,
  selected: false,
  rootClass: [],
};
