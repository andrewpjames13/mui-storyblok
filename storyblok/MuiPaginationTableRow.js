module.exports = {
  name: 'MuiPaginationTableRow',
  display_name: null,
  updated_at: '2020-04-22T14:03:39.670Z',
  schema: {
    rootClass: {
      type: 'options',
      description: '* stroyblok multiselect of css classes\n* Mui Override or extend the styles applied to the component.',
      min_options: '',
      options: [],
    },
    hover: {
      type: 'boolean',
      description: '* mui prop: true | false\n* If true, the table row will shade on hover.',
    },
    selected: {
      type: 'boolean',
      description: '* mui prop: true | false\n* If true, the table row will have the selected shading.',
    },
    content: {
      type: 'bloks',
      restrict_components: true,
      component_whitelist: [
        'MuiTableCell',
      ],
    },
  },
  image: null,
  preview_field: null,
  is_root: false,
  preview_tmpl: null,
  is_nestable: true,
  all_presets: [],
  preset_id: null,
  real_name: 'MuiPaginationTableRow',
  component_group_uuid: null,
};
