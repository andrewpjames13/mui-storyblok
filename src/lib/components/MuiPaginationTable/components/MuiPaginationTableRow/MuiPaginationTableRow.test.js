import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiPaginationTableRow from './MuiPaginationTableRow';

function setup() {
  const props = {
    hover: false,
    selected: false,
    content: [{
      size: 'medium',
      align: 'center',
      content: [{
        align: 'inherit',
        color: 'initial',
        width: '100%',
        height: '100%',
        content: [{
          text: 'Single table row',
          component: 'MuiText',
        }],
        variant: 'body1',
        component: 'MuiTypography',
      }],
      padding: 'default',
      component: 'MuiTableCell',
      sortDirection: 'asc',
    }],
  };
  const comp = shallow(<MuiPaginationTableRow {...props} />);
  return { comp, props };
}

describe('<MuiPaginationTableRow />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(
      <MemoryRouter>
        <MuiPaginationTableRow {...props} />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render MuiTable Row', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });
});
