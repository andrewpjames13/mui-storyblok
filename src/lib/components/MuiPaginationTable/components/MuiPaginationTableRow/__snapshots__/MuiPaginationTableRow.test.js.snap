// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiPaginationTableRow /> snapshot 1`] = `
<tr
  className="MuiTableRow-root makeStyles-root-2"
  role={null}
>
  <td
    aria-sort="ascending"
    className="MuiTableCell-root makeStyles-root-8 MuiTableCell-alignCenter"
  >
    <div
      className="MuiTypography-root makeStyles-root-21 MuiTypography-body1"
    >
      <span
        className="MuiTypography-root makeStyles-root-22 MuiTypography-body1"
      >
         Single table row 
      </span>
    </div>
  </td>
</tr>
`;
